### PR TITLE
docs(example): add component testing example

### PR DESCRIPTION
**Pull request by kentcdodds**
This is *very* much a work in progress. I'm still evaluating whether mocking is what we want to recommend to people. I'm not sure it is. I'm also experimenting with vitest which I mostly like. We'll see where we land.

Making this PR so folks can reference it and provide feedback.

### CODE REVIEW

File: examples/component-testing/app/routes/users/$userId.test.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import UserScreen from "./$userId";
+
+vi.mock("remix", () => {
+  // mock whatever remix APIs you're using
+  return {
+    Form: (props: React.RefAttributes<HTMLFormElement>) => <form {...props} />,
+    useLoaderData: vi.fn(),
+    useLocation: vi.fn(() => ({ key: Math.random().toString(32).slice(2) }))
+  };
+});
```
Comment (MichaelDeBoey) [inline examples/component-testing/app/routes/users/$userId.test.tsx:15]: Tbh, I'm also not sure if we should mock it this way 🤔

I think a custom `render` function that takes care of the contexts and has overridable `action`/`links`/`loader`/`meta` would give the team more control on the whole flow and on how things will be tested/mocked

Comment (kentcdodds) [reply examples/component-testing/app/routes/users/$userId.test.tsx:15]: I agree, it's just a huge challenge to mock the remix context with something sensible. But I do plan on exploring that approach.

**Comment by MichaelDeBoey** [comment]
Just like #2133, I think we can close this one (for now), now that we have #2481 & #372 is closed?

**Comment by kentcdodds** [comment]
Thanks!

**Pull request closed without merging**